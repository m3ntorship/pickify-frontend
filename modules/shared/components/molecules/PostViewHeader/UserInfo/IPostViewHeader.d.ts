import type { IUserInfo } from '../../UserInfo/IUserInfo';

declare namespace IPostViewHeader {
  export interface IProps extends IUserInfo.IData {
    handleEditIconClick: () => void;
    id: string;
  }
  // The IData equals to IUserInfo.IData
}

export { IPostViewHeader };
