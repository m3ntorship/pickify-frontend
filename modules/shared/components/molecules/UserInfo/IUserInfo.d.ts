declare namespace IUserInfo {
  export interface IProps {
    profilePic?: string;
    isHidden: boolean;
    title?: string;
    subTitle: string;
    description?: string;
  }
}

export { IUserInfo };
