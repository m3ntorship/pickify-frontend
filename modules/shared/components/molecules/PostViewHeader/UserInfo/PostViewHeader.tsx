import React from 'react';
import type { ReactElement, FC } from 'react';
import Divider from '../../../atoms/Divider/Divider';
import VerticalThreeDotsIcon from '../../../icons/verticalThreeDots.svg';
import type { IPostViewHeader } from './IPostViewHeader';
import * as EDivider from '../../../atoms/Divider/types/EDivider';
import UserInfo from '../../UserInfo/UserInfo';

const PostViewHeader: FC<IPostViewHeader.IProps> = ({
  profilePic,
  name,
  date,
  handleEditIconClick,
  id,
  isHidden,
}): ReactElement => {
  return (
    <div>
      <div className="flex justify-between items-start pb-s">
        <UserInfo
          isHidden={isHidden}
          profilePic={profilePic}
          name={name}
          date={date}
        />
        <VerticalThreeDotsIcon
          onClick={handleEditIconClick}
          className="fill-grey cursor-pointer"
          data-testid={id}
        />
      </div>
      <Divider type={EDivider.DividerType.Horizontal} length="100%" />
    </div>
  );
};

export default PostViewHeader;
