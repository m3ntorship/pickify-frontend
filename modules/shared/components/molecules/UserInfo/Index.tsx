import React, { useState } from 'react';
import type { ReactElement, FC } from 'react';
import type { IUserInfo } from './IUserInfo';
import Avatar from '../../atoms/avatar/Avatar';
import styles from './UserInfo.module.css';
import {
  humanReadableDate,
  exactDate,
} from '../../../logic/formatDate/FormatDate';

const UserInfo: FC<IUserInfo.IProps> = ({
  size,
  variant,
  imgSrc,
  name,
  date,
}): ReactElement => {
  const [isShown, setIsShown] = useState(false);
  return (
    <div className={styles['outer-wrapper']}>
      <Avatar size={size} variant={variant} imgSrc={imgSrc} />
      <div className={styles['user-wrapper']}>
        <span className={styles.name} data-testid="name">
          {variant === 'anonymous' ? 'Anonymous' : name}
        </span>
        <div
          onMouseEnter={(): void => {
            setIsShown(true);
          }}
          onMouseLeave={(): void => {
            setIsShown(false);
          }}
        >
          {!isShown && (
            <span className={styles.date}>{humanReadableDate(date)}</span>
          )}
          {isShown && <span className={styles.date}>{exactDate(date)}</span>}
        </div>
      </div>
    </div>
  );
};

export default UserInfo;
