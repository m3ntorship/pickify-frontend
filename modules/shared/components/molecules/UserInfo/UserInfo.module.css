.outer-wrapper {
  @apply flex w-min;
}

.user-wrapper {
  @apply max-w-13xl flex flex-col justify-between ml-4 whitespace-nowrap;
}

.title {
  @apply text-sm font-normal block whitespace-nowrap overflow-hidden overflow-ellipsis;
}

.subTitle {
  @apply text-xs font-light	text-dark-grey block;
}
