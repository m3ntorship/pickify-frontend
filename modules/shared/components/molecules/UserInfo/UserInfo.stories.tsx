import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import UserInfo from './UserInfo';
import type { IUserInfo } from './IUserInfo';

export default {
  title: 'molecules/UserInfo',
  component: UserInfo,
  argTypes: {
    profilePic: {
      control: {
        type: 'text',
      },
    },
    title: {
      control: {
        type: 'text',
      },
    },
    subTitle: {
      control: {
        type: 'text',
      },
    },
  },
};

const Template: Story<IUserInfo.IProps> = (args): ReactElement => (
  <UserInfo {...args} />
);
export const UserInfoStory = Template.bind({});
UserInfoStory.args = {
  isHidden: true,
  title: 'Ahmed Ayoub',
  subTitle: '3 months ago',
  description: '13/08/2021 06:09:11',
};
