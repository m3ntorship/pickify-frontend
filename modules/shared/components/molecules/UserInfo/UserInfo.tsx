import React from 'react';
import type { ReactElement, FC } from 'react';
import type { IUserInfo } from './IUserInfo';
import Avatar from '../../atoms/avatar/Avatar';
import styles from './UserInfo.module.css';
import {
  humanReadableDate,
  exactDate,
} from '../../../logic/formatDate/FormatDate';

const UserInfo: FC<IUserInfo.IProps> = ({
  size,
  variant,
  imgSrc,
  name,
  date,
}): ReactElement => {
  return (
    <div className={styles['outer-wrapper']}>
      <Avatar size={size} variant={variant} imgSrc={imgSrc} />
      <div className={styles['user-wrapper']}>
        <span className={styles.name} data-testid="name">
          {variant === 'anonymous' ? 'Anonymous' : name}
        </span>
        <span title={exactDate(date)} className={styles.date}>
          {humanReadableDate(date)}
        </span>
      </div>
    </div>
  );
};

export default UserInfo;
