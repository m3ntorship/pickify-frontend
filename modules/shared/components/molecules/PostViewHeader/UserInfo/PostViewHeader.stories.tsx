import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import PostViewHeader from './Index';
import type { PostViewHeaderTypes } from './PostViewHeaderTypes';

export default {
  title: 'molecules/PostViewHeader',
  component: PostViewHeader,
  argTypes: {
    size: {
      control: {
        type: 'select',
        options: ['large', 'medium', 'small'],
      },
    },
    variant: {
      control: {
        type: 'select',
        options: ['filled', 'notFilled', 'anonymous'],
      },
    },
    imgSrc: {
      control: {
        type: 'text',
      },
    },
  },
};

const Template: Story<PostViewHeaderTypes.IProps> = (args): ReactElement => (
  <PostViewHeader {...args} />
);
export const postViewHeader = Template.bind({});
postViewHeader.args = {
  size: 'medium',
  variant: 'filled',
  name: 'Marzouk el akta3',
  date: '2 hours',
};
