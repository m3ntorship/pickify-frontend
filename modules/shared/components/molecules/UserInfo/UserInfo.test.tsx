import * as React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import UserInfo from './UserInfo';

describe('UserInfo snapshots', () => {
  it('should render avatar with filled variant if there is an image and isHidden={false}', () => {
    const tree = renderer
      .create(
        <UserInfo
          isHidden={false}
          subTitle="3 months ago"
          title="Ahmed Ayoub"
          profilePic="https://placeimg.com/640/480/any"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render avatar with filled variant event if isHidden={true} but there is an image', () => {
    const tree = renderer
      .create(
        <UserInfo
          isHidden
          subTitle="3 months ago"
          title="Ahmed Ayoub"
          profilePic="https://placeimg.com/640/480/any"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render avatar with anonymous variant if isHidden={true} with no image', () => {
    const tree = renderer
      .create(<UserInfo isHidden subTitle="3 months ago" title="Ahmed Ayoub" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render avatar with notFilled variant if isHidden={false} with no image', () => {
    const tree = renderer
      .create(
        <UserInfo
          isHidden={false}
          subTitle="3 months ago"
          title="Ahmed Ayoub"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('UserInfo RTL', () => {
  it('should display {Anonymous} as title if the post owner choose to hide his identity', () => {
    render(<UserInfo isHidden subTitle="3 months ago" />);
    expect(screen.getByTestId('title')).toHaveTextContent('Anonymous');
  });

  it('should display {Anonymous} as well if no title has passed and isHidden={true}', () => {
    render(<UserInfo isHidden subTitle="3 months ago" />);
    expect(screen.getByTestId('title')).toHaveTextContent('Anonymous');
  });

  it('should display a title if it is passed even if isHidden={ture}', () => {
    render(<UserInfo isHidden subTitle="3 months ago" title="Ahmed Ayoub" />);
    expect(screen.getByTestId('title')).toHaveTextContent('Ahmed Ayoub');
  });

  it('should display a title if it is passed and isHidden={false}', () => {
    render(
      <UserInfo isHidden={false} subTitle="3 months ago" title="Ahmed Ayoub" />,
    );
    expect(screen.getByTestId('title')).toHaveTextContent('Ahmed Ayoub');
  });

  it('should display a title if it is passed and isHidden={false}', () => {
    render(
      <UserInfo isHidden={false} subTitle="3 months ago" title="Ahmed Ayoub" />,
    );
    expect(screen.getByTestId('title')).toHaveTextContent('Ahmed Ayoub');
  });
});
