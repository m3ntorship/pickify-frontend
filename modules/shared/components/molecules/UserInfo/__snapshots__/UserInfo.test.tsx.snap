// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo snapshots should render avatar with anonymous variant if isHidden={true} with no image 1`] = `
<div
  className="outer-wrapper"
>
  <div
    className="wrapper medium"
  >
    <svg
      className="absolute w-full h-full"
      data-file-name="SvgAnonymous"
      height={40}
      onClick={[Function]}
      width={40}
    />
  </div>
  <div
    className="user-wrapper"
  >
    <span
      className="title"
      data-testid="title"
    >
      Ahmed Ayoub
      <span
        className="ml-2 text-grey text-xs"
      >
        (anonymous)
      </span>
    </span>
    <span
      className="subTitle"
    >
      3 months ago
    </span>
  </div>
</div>
`;

exports[`UserInfo snapshots should render avatar with filled variant event if isHidden={true} but there is an image 1`] = `
<div
  className="outer-wrapper"
>
  <div
    className="wrapper medium"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        className="absolute w-full h-full"
        decoding="async"
        onClick={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="user-wrapper"
  >
    <span
      className="title"
      data-testid="title"
    >
      Ahmed Ayoub
      <span
        className="ml-2 text-grey text-xs"
      >
        (anonymous)
      </span>
    </span>
    <span
      className="subTitle"
    >
      3 months ago
    </span>
  </div>
</div>
`;

exports[`UserInfo snapshots should render avatar with filled variant if there is an image and isHidden={false} 1`] = `
<div
  className="outer-wrapper"
>
  <div
    className="wrapper medium"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        className="absolute w-full h-full"
        decoding="async"
        onClick={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="user-wrapper"
  >
    <span
      className="title"
      data-testid="title"
    >
      Ahmed Ayoub
    </span>
    <span
      className="subTitle"
    >
      3 months ago
    </span>
  </div>
</div>
`;

exports[`UserInfo snapshots should render avatar with notFilled variant if isHidden={false} with no image 1`] = `
<div
  className="outer-wrapper"
>
  <div
    className="wrapper medium"
  >
    <svg
      className="absolute w-full h-full"
      data-file-name="SvgNotFilled"
      height={40}
      onClick={[Function]}
      width={40}
    />
  </div>
  <div
    className="user-wrapper"
  >
    <span
      className="title"
      data-testid="title"
    >
      Ahmed Ayoub
    </span>
    <span
      className="subTitle"
    >
      3 months ago
    </span>
  </div>
</div>
`;
